src/core/account_builder.cpp

```diff
@@ -0,0 +1,278 @@
+        {
+            zip = 0;
+        }
+        catch (const std::out_of_range &err)
+        {
+            zip = 0;
+        }
+
+        try
+        {
+            address = Address(account_field.at(current_street), account_field.at(current_city), account_field.at(current_state), zip.value());
+        }
+        catch (const chocan_user_exception &err)
+        {
+            issues.emplace(err);
+
+        }
+    }
+
+    if (!issues)
+        build_phase.pop();
+
+    return;
+}
+
+const chocan_user_exception Account_Builder::get_current_issues()
+{
+
+    std::optional<chocan_user_exception> return_msg = issues;
+
+    issues.reset();
+
+    return return_msg.value_or(chocan_user_exception{"", {}});
+}
+
+const std::string Account_Builder::get_current_field() const
+{
+
+    if (build_phase.empty())
+        return review_account();
+
+    return build_phase.top();
+}
+
+void Account_Builder::parseName(const std::string &input)
+{
+
+    size_t delimiter_position = input.find(",");
+
+    if(delimiter_position == std::string::npos){
+    
+        account_field[current_first] = "";
+        account_field[current_last] = "";
+
+    }else{
+
+        account_field[current_first] = input.substr(0, input.find(","));
+        account_field[current_last] = input.substr(input.find(",") + 1, input.length());
+
+        remove_leading_white_space(account_field[current_first]);
+        remove_leading_white_space(account_field[current_last]);
+    }
+
+}
+
+void Account_Builder::parseAddress(const std::string &input)
+{
+
+    size_t delimiter_position = 0;
+    const std::string delimiter = ",";
+    std::string temp_input(input + delimiter);
+    std::stack<field_types> current_field({current_zip, current_state, current_city, current_street});
+
+    while (!current_field.empty() && (delimiter_position = temp_input.find(delimiter)) != std::string::npos)
+    {
+        account_field[current_field.top()] = temp_input.substr(0, delimiter_position);
+
+        remove_leading_white_space(account_field[current_field.top()]);
+
+        current_field.pop();
+
+        temp_input.erase(0, delimiter_position + delimiter.length());
+    }
+
+    while(!current_field.empty()){
+        account_field[current_field.top()] = "";
+        current_field.pop();
+    }
+
+    account_field[current_zip] = account_field[current_zip].substr(0,5);
+}
+
+const std::string Account_Builder::valid_account_type(const std::string &input)
+{
+
+    std::string temp("");
+
+    for (std::vector<std::string>::iterator it = valid_types.begin(); it <= valid_types.end(); ++it)
+    {
+
+        if (input == *it)
+        {
+
+            temp = *it;
+            temp[0] = toupper(temp[0]);
+            it = valid_types.end();
+        }
+    }
+
+    remove_leading_white_space(temp);
+    
+    return temp;
+}
+
+const std::string Account_Builder::review_account() const
+{
+
+    return "(Y)es or (N)o to confirm/reject this account."
```

Comment:
Also, this again breaks a big design principle. You're account builder is explicitly handling responsibilities of the presentation layer by formatting strings for console output. The domain layer should have no knowledge that it is being used as a console application.

Reply:
The logic behind confirmation or not should be handled in a different application state which will also have a different application view.

Reply:
yup, i see what you mean. this method will be gone.

Reply:
that method is gone